src/micromamba/config.cpp

```diff
@@ -88,6 +185,422 @@ set_config_describe_command(CLI::App* subcom)
     });
 }
 
+void
+set_config_path_command(CLI::App* subcom)
+{
+    auto& config = Configuration::instance();
+
+    auto& system_path = config.insert(Configurable("config_set_system_path", false)
+                                          .group("cli")
+                                          .description("Set configuration on system's rc file"),
+                                      true);
+    subcom->add_flag("--system", system_path.set_cli_config(0), system_path.description());
+
+    auto& env_path = config.insert(Configurable("config_set_env_path", false)
+                                       .group("cli")
+                                       .description("Set configuration on env's rc file"),
+                                   true);
+    subcom->add_flag("--env", env_path.set_cli_config(0), env_path.description());
+
+    auto& file_path = config.insert(Configurable("config_set_file_path", fs::path())
+                                        .group("cli")
+                                        .description("Set configuration on system's rc file"),
+                                    true);
+    subcom->add_option("--file", file_path.set_cli_config(fs::path()), file_path.description());
+}
+
+void
+set_config_prepend_command(CLI::App* subcom)
+{
+    set_config_path_command(subcom);
+
+    auto& config = Configuration::instance();
+    auto& ctx = Context::instance();
+
+    auto& prepend_map = config.insert(
+        Configurable("prepend_map", std::vector<std::string>({ "" }))
+            .group("Output, Prompt and Flow Control")
+            .description("Add one configuration value to the beginning of a list key"));
+    subcom->add_option(
+        "prepend_map", prepend_map.set_cli_config({ "" }), prepend_map.description());
+
+    auto& file_path = config.at("config_set_file_path").get_wrapped<fs::path>();
+    auto& env_path = config.at("config_set_env_path").get_wrapped<bool>();
+    auto& system_path = config.at("config_set_system_path").get_wrapped<bool>();
+
+    subcom->callback([&]() {
+        config.at("use_target_prefix_fallback").set_value(true);
+        config.at("show_banner").set_value(false);
+        config.at("target_prefix_checks")
+            .set_value(MAMBA_ALLOW_EXISTING_PREFIX | MAMBA_ALLOW_MISSING_PREFIX
+                       | MAMBA_ALLOW_NOT_ENV_PREFIX | MAMBA_NOT_EXPECT_EXISTING_PREFIX);
+        config.load();
+
+        fs::path rc_source = env::expand_user(env::home_directory() / ".condarc");
+        std::ofstream rc_file;
+
+        std::string prepend_key = prepend_map.value().front();
+        std::string prepend_value = prepend_map.value().back();
+
+        if (file_path.configured())
```

Comment:
there is a bit of code-duplication going on now. Do you think we can refactor that into a single callback? You can move the code that gets the config object in the callback, too.

You could have a function that is called:

```
enum CONFIG_MOD_ACTION { SET, APPEND, PREPEND };

void modify_config(CONFIG_MOD_ACTION action)
{
    auto& config = Config::instance();
    config.load();
    auto& filepath = config.at("config_set_file_path").get_wrapped<fs::path>();
     ...
}

// and then in the callbacks:

subcom->callback([&]() {
    modify_config(CONFIG_MOD_ACTION::SET);
}
...
```

Reply:
Sorry, I just saw the comment now.
Yeah, there's a bunch of code duplication in these methods.